### PR TITLE
Move Rx classes under version namespace to prevent collisions

### PR DESCRIPTION
**Pull request by chenthorne**
For Issue #42

Also updates Gradle and related Android plugin; fixes linter errors on sample projects.

### CODE REVIEW

File: config/quality_android.gradle [hunk 1]
```diff
@@ -40,4 +46,4 @@ task pmd(type: Pmd) {
   }
 }
 
-check.dependsOn 'findbugs', 'pmd'
+check.dependsOn 'assemble', 'findbugs', 'pmd'
```
Comment (chenthorne) [inline config/quality_android.gradle]: Needed for `check` to work from `clean`

File: config/quality_android.gradle [hunk 2]
```diff
@@ -1,6 +1,12 @@
 apply plugin: 'findbugs'
 apply plugin: 'pmd'
 
+android {
+  lintOptions {
+    disable "GoogleAppIndexingWarning", "AllowBackup"
```
Comment (chenthorne) [inline config/quality_android.gradle]: Lint errors that do not need to be alerted about for sample projects.

File: grox-sample-rx/build.gradle [hunk 3]
```diff
@@ -2,11 +2,16 @@ apply plugin: 'com.android.application'
 apply from: rootProject.file("${quality_gradle_android_file}")
 apply plugin: 'com.github.hierynomus.license'
 
+repositories {
+  google()
```
Comment (chenthorne) [inline grox-sample-rx/build.gradle]: Needed when updating new AGP

**Comment by ingerul93** [comment]
Please check my PR to solve license issue.
https://github.com/groupon/grox/pull/44

**Comment by alin-turcu** [comment]
@chenthorne you need to update .travis.yml also. That's why the build fails

**Comment by gracefulife** [comment]
👍

**Pull request merged**